session.go

```diff
@@ -973,24 +973,71 @@ func (st *StmtTxn) SeekReverse(k kv.Key) (kv.Iterator, error) {
 	return kv.NewUnionIter(bufferIt, retrieverIt, true)
 }
 
-// StmtCommit implements the Transaction interface.
-func (st *StmtTxn) StmtCommit() error {
-	err := kv.WalkMemBuffer(st.buf, func(k kv.Key, v []byte) error {
+// StmtCommit implements the context.Context interface.
+func (s *session) StmtCommit() error {
+	st := &s.StmtTxn
+	err := kv.WalkMemBuffer(s.buf, func(k kv.Key, v []byte) error {
 		if len(v) == 0 {
 			return errors.Trace(st.Transaction.Delete(k))
 		}
 		return errors.Trace(st.Transaction.Set(k, v))
 	})
 	st.buf.Reset()
+
+	if err == nil {
+		// Need to flush binlog.
+		for key, value := range st.mutations {
+			mutation := getBinlogMutation(s, key)
+			mergeToMutation(mutation, value)
+			delete(st.mutations, key)
```

Comment:
what about a new map after flush loop?

Reply:
That make much more unnecessary allocation.
When binlog is not enabled, current code makes no allocation at all!

Reply:
but if we check the len(st.mutations) > 0 then do this, what about?

Reply:
Anyway, you have to:
``
st.mutations = nil 
st.mutations = make(map[int64]*binlog.TableMutations
``